scheduler.rst

```diff
@@ -0,0 +1,328 @@
+
+As mentioned, this component, even if it's an independent component, it draws its foundation and inspiration from the Messenger component.
+
+On one hand, it adopts well-established concepts from Messenger (such as message, handler, bus, transport, etc.).
+For example, the task of creating a report is considered as a message that will be directed, and processed by the corresponding handler.
+
+However, unlike Messenger, the messages will not be dispatched in the first instance. Instead, the aim is to create them based on a predefined frequency.
+
+This is where the specific transport in Scheduler, known as the :class:`Symfony\\Component\\Scheduler\\Messenger\\SchedulerTransport`, plays a crucial role.
+The transport autonomously generates directly various messages according to the assigned frequencies.
+
+From (Messenger cycle):
+
+.. image:: /_images/components/messenger/basic_cycle.png
+    :alt: Symfony Messenger basic cycle
+
+To (Scheduler cycle):
+
+.. image:: /_images/components/scheduler/scheduler_cycle.png
+    :alt: Symfony Scheduler basic cycle
+
+In essence, it is crucial to precisely define the message to be conveyed and processed. In Scheduler, the concept of a message takes on a very particular characteristic;
+it should be recurrent: It's a :class:`Symfony\\Component\\Scheduler\\RecurringMessage`.
+
+Clarifying the need to define and attach RecurringMessages to a Schedule
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In order to generate various messages based on their defined frequencies, configuration is necessary.
+The heart of the scheduling process and its configuration resides in a class that must extend the :class:`Symfony\\Component\\Scheduler\\ScheduleProviderInterface`.
+
+The purpose of this provider is to return a schedule through the method :method:`Symfony\\Component\\Scheduler\\ScheduleProviderInterface::getMethod` containing your different recurringMessages.
+
+The :class:`Symfony\\Component\\Scheduler\\Attribute\\AsSchedule` attribute, which by default references the ``default`` named schedule, allows you to register on a particular schedule::
+
+    // src/Scheduler/MyScheduleProvider.php
+    namespace App\Scheduler;
+
+    class MyScheduleProvider implements ScheduleProviderInterface
+    {
+        public function getSchedule(): Schedule
+        {
+            // ...
+        }
+    }
+
+.. tip::
+
+    This becomes important when initiating the ``messenger:consume`` command, especially when specifying one or more specific transports.
+    In Scheduler, the transport is named ``scheduler_nameofyourschedule``.
+
+.. tip::
+
+    It is a good practice to memoize your schedule to prevent unnecessary reconstruction if the ``getSchedule`` method is checked by another service or internally within Symfony
+
+
+The Concept of RecurringMessage in Scheduler
```

Comment:
Are there any other types of messages that you would schedule? Suggestion for the title:

> Scheduling Recurring Messages

And, I like this section. This is where we get into the meat of things